File: 2026/08-August/14-08-26/0003-longest-substring-without-repeating-characters.py
"""
LeetCode 3 · Longest Substring Without Repeating Characters · Medium
https://leetcode.com/problems/longest-substring-without-repeating-characters/

Pattern : Sliding Window
Solved  : 14 Aug 2026
Time    : O(n)
Space   : O(1)
"""

#3.  Valid Anagram
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        w = list(t)
        for q in s:
            if q in w:
                ind = w.index(q)
                w.pop(ind)
            else:
                return False
        if not w:
            return True
        else:
            return False

File: 2026/02-February/01-02-26/0003-longest-substring-without-repeating-characters.cpp
/*
 * LeetCode 3 · Longest Substring Without Repeating Characters · Medium
 * https://leetcode.com/problems/longest-substring-without-repeating-characters/
 *
 * Pattern : Sliding Window
 * Solved  : 01 Feb 2026
 * Time    : O(N) or O(N^2) depending on find/erase implementation, Space: O(N)
 * Space   : O(N)
 */

// LeetCode 3 - Longest Substring Without Repeating Characters
// Solved on: 2026-02-01
// Topic: String
// Approach: Sliding Window
// Time: O(N) or O(N^2) depending on find/erase implementation, Space: O(N)

class Solution {
public:
    int lengthOfLongestSubstring(string s) {
         int count =0;
         string ans="";
         int m=0;
         for(int i=0;i<s.size();i++){
            size_t temp=ans.find(s[i]);
            if(temp==string::npos){
                ans+=s[i];
                
            }
            else{
                
                ans.erase(0,temp+1);
                
                
                ans+=s[i];
                
               


            }
            count=ans.size();
            m=max(m,count);
        }
        
        return m;
    }
};
